pkg/controller/template/render.go

```diff
@@ -33,15 +34,15 @@ const (
 	filesDir = "files"
 	unitsDir = "units"
 
-	// TODO: these constants are wrong, they should match what is reported by the infrastructure provider
-	platformAWS       = "aws"
-	platformAzure     = "azure"
-	platformBaremetal = "baremetal"
-	platformOpenstack = "openstack"
-	platformLibvirt   = "libvirt"
-	platformNone      = "none"
-	platformVSphere   = "vsphere"
-	platformBase      = "_base"
+	platformAWS       = string(configv1.AWSPlatformType)
+	platformAzure     = string(configv1.AzurePlatformType)
+	platformOpenstack = string(configv1.OpenStackPlatformType)
+	platformLibvirt   = string(configv1.LibvirtPlatformType)
+	platformNone      = string(configv1.NonePlatformType)
+	platformVSphere   = string(configv1.VSpherePlatformType)
+	platformBareMetal = string(configv1.BareMetalPlatformType)
```

Comment:
we can get rid of these assignements and just use `string(...)`  in the switch cases below right?

Reply:
sure, just did this for readability because the vars get used a few times